packages/remix-node/upload/fileUploadHandler.ts

```diff
@@ -88,22 +92,32 @@ export function createFileUploadHandler({
       return;
     }
 
-    let filedir = resolvePath(
+    let dir =
       typeof directory === "string"
         ? directory
-        : directory({ filename, encoding, mimetype })
-    );
+        : directory({ filename, encoding, mimetype });
 
-    let filepath = resolvePath(
-      filedir,
-      typeof file === "string" ? file : file({ filename, encoding, mimetype })
-    );
+    if (!dir) {
+      stream.resume();
+      return;
+    }
+
+    let filedir = resolvePath(dir);
+    let path =
+      typeof file === "string" ? file : file({ filename, encoding, mimetype });
+
+    if (!path) {
+      stream.resume();
+      return;
+    }
+
+    let filepath = resolvePath(filedir, path);
 
     if (avoidFileConflicts) {
       filepath = await uniqueFile(filepath);
     }
 
-    await mkdir(dirname(filepath), { recursive: true });
+    await mkdir(dirname(filepath), { recursive: true }).catch(() => {});
```

Comment:
I thought we agreed to just let this thing throw?

Reply:
It throws if the directory exists, that's the issue. The message from the file creation will mention directory doesn't exist so we are good.